### ISSUE TITLE
dlang-bot not closing all referenced issues

### ISSUE THREAD

**Issue by aG0aep6G** [dlang-bots/dlang-bot#175]
In <https://github.com/dlang/phobos/pull/6056>, the bot picked up 4 issues, but it closed only one. Bug?

See also: https://forum.dlang.org/post/rqzylqoeeptjbdgmwbbw@forum.dlang.org

**Comment by wilzbach** [comment]
tl;dr: "hard to fix", but definitely a bug :/

Copy/pasted my response from the PR

---

A bit of explanation:

The bugs are closed by the Bugzilla integration - not the bot.
The changelog is generated by https://github.com/dlang/tools/blob/master/changed.d - it parses the git log between two revisions with the same regex as the bot and the bugzilla integration does. However, it then queries Bugzilla and filters for only referenced bugs which are closed.
So the bug overview by the bot is merely a help for reviewers as (a) it provides a handy link to the references issues and (b) allows them to verify that the issues have been referenced correctly.
See also: https://github.com/dlang-bots/dlang-bot#automated-references

Now to the bug - it looks like the Bugzilla integration is a bit dumber and only picks up one bug regex per commit. There's much I can do about (except for writing our own Bugzilla integration).
In any case, a "workaround" for the future if you have to use one commit is to include multiple issues in one regex, e.g. "Fixes issues 17494, 17505, 17506".

---

**Comment by MartinNowak** [comment]
We could do many things about this ;), e.g. file a PR with github's bugzilla integration or modify the bot to also only use the first match.
But as said, right now GH only picks up the first match, not all of them.
https://github.com/dlang-bots/dlang-bot#referencing-multiple-issues
